crates/matrix-sdk-ui/src/timeline/builder.rs

```diff
@@ -197,7 +209,9 @@ impl TimelineBuilder {
         ];
 
         let (msg_sender, msg_receiver) = mpsc::channel(1);
-        spawn(send_queued_messages(inner.clone(), room.clone(), msg_receiver));
+        if !read_only {
+            spawn(send_queued_messages(inner.clone(), room.clone(), msg_receiver));
```

Comment:
Could be sender/receiver be created only when `!read_only`? (would require making the `msg_sender` `Option`al in the Timeline)

Reply:
Sure, but what do we gain from having a `None` instead of a closed sender?

Reply:
Mostly better understanding of the code; when the sender is not used, it's `None` instead of existing and being unused.

Reply:
Well, it can still be used in that case. It will just be an error (see updated error logs in `timeline/mod.rs`).

Reply:
Yeah, my comment probably implied just a bit more work, to also guard the methods that send messages and make sure they double-check for the presence of the sender first. But it's good enough the way it is right now :+1: